pkg/registry/apis/provisioning/jobs/syncer.go

```diff
@@ -297,7 +326,12 @@ func (r *Syncer) replicateTree(ctx context.Context, ref string) error {
 		info.Hash = entry.Hash
 		info.Modified = nil // modified?
 
-		if err := r.replicateFile(ctx, info); err != nil {
+		parsed, err := r.parseResource(ctx, info)
```

Comment:
I moved out the `parseResource` because the parsed resource is used by `deleteFile` and `replicateFile` (I think it's a better distribution of responsibilities).